src/js/jsx/search/SearchBar.jsx

```diff
@@ -500,20 +500,29 @@ define(function (require, exports, module) {
 
         render: function () {
             var searchStrings = nls.localize("strings.SEARCH"),
-                noMatchesOption = {
+                filters = this.state.filter,
+                placeholderText,
+                clearInputBtn;
+
+            var flux = this.getFlux(),
+                searchStore = flux.store("search"),
+                registeredSearchModules = searchStore._registeredSearchTypes;
+
+            var lastFilter = filters[filters.length - 1];
+
+            var searchPayload = registeredSearchModules[lastFilter];
+            var noOptionsString = searchPayload ? searchPayload.noOptionsString : null;
```

Comment:
Sometimes the payload returns null ie. in the case of Layers. There is one registered search module for layers, but then we break down the categories into Pixel Layers, Graphic Layers, etc. The filter "Pixel" is a valid one but there is no specific registered module for just Pixel Layers. Therefore, the `searchPayload` would return null and that is why that line is there to check for the existence of the payload and then line 517 is there to check for the existence of the noOptionsString.